docs/Using-Virtual-Environment.md

```diff
@@ -1,33 +1,52 @@
 # Using Virtual Environment
 
 ## What is a Virtual Environment?
-A Virtual Environment is a walled garden for Python projects.  
+A Virtual Environment is a self contained directory tree that contains a Python installation 
+for a particular version of Python, plus a number of additional packages.  
 
 ## Why should I use a Virtual Environment?
 A Virtual Environment keeps all dependencies for the project separate from dependencies 
 of other projects. This has a few advantages:
-1. Firstly, it makes dependency management for the project easy.
-1. Secondly, it enables using and testing of different library versions by quickly 
+1. It makes dependency management for the project easy.
+1. It enables using and testing of different library versions by quickly 
 spinning up a new environment and verifying the compatibility of the code with the
 different version. 
 
 Requirement - Python 3.6 must be installed on the machine you would like 
 to run ML-Agents on (either local laptop/desktop or remote server). Python 3.6 can be 
 installed from [here](https://www.python.org/downloads/). 
 
+
+## Installing Pip (Required)
+
+1. Download the `get-pip.py` file using the command `curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py`
+1. Run the following `python3 get-pip.py`
+1. Check pip version using `pip3 -V`
+
+Note (for Ubuntu users): If the `ModuleNotFoundError: No module named 'distutils.util'` error is encountered, then
+python3-distutils needs to be installed. Install python3-distutils using `sudo apt-get install python3-distutils` 
+
 ## Mac OS X Setup
 
-1. Create a folder where the virtual environments will live ` $ mkdir ~/python-venvs `
-1. To create a new environment named `test-env` execute `$ python3 -m venv ~/python-envs/test-env`  
-1. To activate the environment execute `$ source ~/python-envs/test-env/bin/activate`
+1. Create a folder where the virtual environments will reside `$ mkdir ~/python-venvs`
+1. To create a new environment named `sample-env` execute `$ python3 -m venv ~/python-envs/sample-env`  
+1. To activate the environment execute `$ source ~/python-envs/sample-env/bin/activate`
+1. Verify pip version is the same as in the __Installing Pip__ section. In case it is not the latest, upgrade to 
+the latest pip version using `pip3 install --upgrade pip` 
 1. Install ML-Agents package using `$ pip3 install mlagents`
-1. To deactivate the environment execute `$ deactivate `
+1. To deactivate the environment execute `$ deactivate`
 
 ## Ubuntu Setup 
 
 1. Install the python3-venv package using `$ sudo apt-get install python3-venv`
-1. Follow steps 2-5 in the Mac OS X installation.
+1. Follow the steps in the Mac OS X installation.
 
 ## Windows Setup
 
-Coming Soon.
\ No newline at end of file
+1. Create a folder where the virtual environments will reside `$ md python-venvs`
```

Comment:
my last suggestion - might consider linking directly to step 2 here instead https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Installation-Windows.md

Reply:
I'm not clear about your last suggestion. As mentioned in the google doc posted on slack we are removing the Installation-Windows.md page as we are proposing a platform agnostic way to install ml-agents.

Reply:
hey - sorry, i think i missed that doc.  that actually makes a lot more sense!